### PR TITLE
Beam job util

### PR DESCRIPTION
**Pull request by beczkowb**
Adding `BeamJob` which takes pipeline options and driver, then enriches the pipeline options and runs the job.

### CODE REVIEW

File: bigflow/dataflow.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+            driver_arguments: typing.Optional[dict] = None,
+            wait_until_finish: bool = True):
+        self._id = id
+        self.driver_callable = driver_callable
+        self.driver_arguments = driver_arguments
+        self.pipeline_options = pipeline_options
+        self.wait_until_finish = wait_until_finish
+
+    def execute(self, context: JobContext):
+        pipeline_options = self._apply_logging(self.pipeline_options, context.workflow.workflow_id)
```
Comment (iwko) [inline bigflow/dataflow.py:24]: _apply_labels?

**Comment by iwko** [comment]
Conflicts with master

File: bigflow/dataflow.py (hunk 1 above)
Comment (anjensan) [inline bigflow/dataflow.py:24]: JobContext.worfkflow is `Optaionl[Workflow]`
Add 'if context.workflow' or we can change type of JobContext.workflow, which might be tricky however.

File: bigflow/__init__.py [hunk 2]
```diff
@@ -26,3 +26,9 @@
     __all__.append('monitoring')
 except ImportError:
     pass
+
+try:
+    from . import dataflow
+    __all__.append('dataflow')
```
Comment (anjensan) [inline bigflow/__init__.py:32]: IMO we should not add 'dataflow' to `__all__`
It is required only to support cases like:

```python
from bigflow import *
my_job = dataflow.BeamJob(...)
```

it is better to write
```python
import bigflow
my_job = bigflow.dataflow.BeamJob(...)
```
but the best option is to explicitly import needed module:

```python
from bigflow.dataflow import BeamJob
my_job = BeamJob(...)
```
or
```python
import bigflow.dataflow
my_job = bigflow.dataflow.BeamJob(...)
```

In such case there is no need to add anything to `bigflow/__init__.py` at all

File: bigflow/dataflow.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        pipeline = Pipeline(options=pipeline_options)
+        self.driver_callable(pipeline, context, **self.driver_arguments)
+        pipeline = pipeline.run()
+        if self.wait_until_finish:
+            pipeline.wait_until_finish()
+
+    @property
```
Comment (anjensan) [inline bigflow/dataflow.py:31]: There is no real need to add this property

File: bigflow/dataflow.py [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+from apache_beam.options.pipeline_options import PipelineOptions, GoogleCloudOptions
+
+from .workflow import Job, JobContext
+
+
+class BeamJob(Job):
+    def __init__(
+            self,
+            id: str,
+            driver_callable: typing.Callable[[Pipeline, JobContext, typing.Dict], None],
```
Comment (anjensan) [inline bigflow/dataflow.py:13]: This function is used as `f(pipeline, context, **kwargs)`, not as `f(pipeline, context, kwargs)`
AFAIK in python 3.7 there is no way to express such callback type (with **kwargs)

**Review by anjensan — CHANGES_REQUESTED** [review]

File: bigflow/dataflow.py (hunk 1 above)
Comment (beczkowb) [reply bigflow/dataflow.py:24]: We apply labels to apply logging, so it's more meaningful to say "apply_logging".

File: bigflow/dataflow.py (hunk 4 above)
Comment (beczkowb) [reply bigflow/dataflow.py]: switched from kwargs to a dict argument

File: bigflow/dataflow.py (hunk 3 above)
Comment (beczkowb) [reply bigflow/dataflow.py]: true

File: bigflow/dataflow.py (hunk 1 above)
Comment (beczkowb) [reply bigflow/dataflow.py]: done

File: bigflow/__init__.py (hunk 2 above)
Comment (beczkowb) [reply bigflow/__init__.py]: done

**Review by anjensan — APPROVED** [review]

**Pull request merged**